### DISCUSSION TITLE
Can optuna dashboard load the study persisted in .pkl file?

### DISCUSSION THREAD

**Discussion by ttzhu813** [optuna/optuna-dashboard#813] [Q&A] [answered]
Hi there, can optuna dashboard load the study persisted in .pkl file? I only found user guides about RDBStorage and JournalFileStorage.

**Comment by c-bata** [comment]
Thank you for using Optuna. Unfortunately, Optuna Dashboard doesn't support such a use case. Following code might work though.

```python
import pickle
from optuna_dashboard import run_server

study = pickle.loads("...pkl")
storage = study.storage

run_server(storage)
```

https://optuna-dashboard.readthedocs.io/en/latest/getting-started.html#python-interface

**Reply by ttzhu813** [reply]
That works for me perfectly, thanks a lot~

**Reply by ttzhu813** [reply]
Btw, can I load multiple studies at a time using this method you mentioned?

**Reply by c-bata** [reply, marked as answer]
@ttzhu813 Could you try the following code?

```python
import pickle
import optuna
from optuna_dashboard import run_server

pickle_objects = ["foo.pkl", "bar.pkl"]

storage = optuna.storages.InMemoryStorage()
for p in pickle_objects:
    study = pickle.loads(p)
    optuna.copy_study(
        from_study_name=study.study_name,
        from_storage=study.storage,
        to_storage=storage,
    )

run_server(storage)
```

**Reply by ttzhu813** [reply]
Thanks a lot!
